docs/kap_proposals/kap_11_hashicorp_vault_transit.md

```diff
@@ -0,0 +1,151 @@
+
+```shell
+$ echo "any.value:whatever-you_may*like" > somefile.txt
+$  kapitan refs --write vaulttransit:<target_name>/to/secret_inside_kapitan --file somefile.txt --target <target_name>
+```
+or in a single line
+```shell
+$ echo "any.value:whatever-you_may*like"  | kapitan refs --write vaulttransit:<target_name>/to/secret_inside_kapitan -t <target_name> -f -
+```
+The entire string __"any.value:whatever-you_may*like"__ will be encrypted by vault and looks like this in return: `vault:v2:Jhn3UzthKcJ2s+sEiO60EUiDmuzqUC4mMBWp2Vjg/DGl+GDFEDIPmAQpc5BdIefkplb6yrJZq63xQ9s=`. This then gets base64 encoded and stored in the secret_inside_kapitan. Now secret_inside_kapitan contains the following
+
+```yaml
+data: dmF1bHQ6djI6SmhuM1V6dGhLY0oycytzRWlPNjBFVWlEbXV6cVVDNG1NQldwMlZqZy9ER2wrR0RGRURJUG1BUXBjNUJkSWVma3BsYjZ5ckpacTYzeFE5cz0=
+encoding: original
+type: vaulttransit
+vault_params:
+  VAULT_ADDR: http://127.0.0.1:8200
+  VAULT_SKIP_VERIFY: 'True'
+  VAULT_TOKEN: s.i53a1DL83REM61UxlJKLdQDY
+  auth: token
+  crypto_key: key
+  mount: transit
+  always_latest: false
+```
+
+Encoding tells the type of data given to kapitan, if it is `original` then after decoding base64 we'll get the original secret and if it is `base64` then after decoding once we still have a base64 encoded secret and have to decode again.
+Parameters in the secret file are collected from the inventory of the target we gave from CLI `--target my_target`. If target isn't provided then kapitan will identify the variables from the environment, but providing `auth` is necessary as a key inside target parameters like the one shown:
+```yaml
+parameters:
+  kapitan:
+    vars:
+      target: my_target
+      namespace: my_namespace
+    secrets:
+      vaulttransit:
+        VAULT_ADDR: http://vault.example.com:8200
+        VAULT_TOKEN: s.i53a1DL83REM61UxlJKLdQDY
+        VAULT_SKIP_VERIFY: "True"
+        auth: token
+        mount: transit
+        crypto_key: new_key   
+        always_latest: False
+```
+Environment variables that can be defined in kapitan inventory are `VAULT_ADDR`, `VAULT_NAMESPACE`, `VAULT_SKIP_VERIFY`, `VAULT_CLIENT_CERT`, `VAULT_CLIENT_KEY`, `VAULT_CAPATH` & `VAULT_CACERT`.
+Extra parameters that can be defined in inventory are:
+* `auth`: specify which authentication method to use like `token`,`userpass`,`ldap`,`github` & `approle`
+* `mount`: specify the mount point of key's path. e.g if path=`alpha-secret/foo/bar` then `mount: alpha-secret` (default `secret`)
+* `crypto_key`: Name of the `encryption key` defined in vault
+* `always_latest`: Always rewrap ciphertext to latest rotated `crypto_key` version  
+Environment variables cannot be defined in inventory are `VAULT_TOKEN`,`VAULT_USERNAME`,`VAULT_PASSWORD`,`VAULT_ROLE_ID`,` VAULT_SECRET_ID`.
```

Comment:
VAULT_TOKEN and other shows up in the inventory examples above, however I agree with this statement but am confused

Reply:
I would change this to:

> Environment variables **should NOT** be defined in inventory are ...